autopush/websocket.py

```diff
@@ -352,63 +346,40 @@ def process_hello(self, data):
         # This must be a helo, or we kick the client
         cmd = data.get("messageType")
         if cmd != "hello":
-            self.sendClose()
-            return
+            return self.sendClose()
 
         if self.uaid:
-            self.returnError("hello", "duplicate hello", 401)
-            return
+            return self.returnError("hello", "duplicate hello", 401)
 
         uaid = data.get("uaid")
-        valid = False
-        if uaid:
-            try:
-                valid = bool(uuid.UUID(uaid))
-            except ValueError:
-                pass
-        if not uaid or not valid:
-            uaid = str(uuid.uuid4())
+        _, uaid = validate_uaid(uaid)
         self.uaid = uaid
 
-        connect = data.get("connect")
-        if connect and self.ap_settings.bridge:
-            self.transport.pauseProducing()
-            d = self.deferToThread(self.ap_settings.bridge.register, uaid,
-                                   connect)
-            d.addCallback(self._check_router, True)
-            d.addErrback(self.err_hello)
-        else:
-            self._check_router(False)
+        # Default router choice
+        router_type = data.get("router_type", "simplepush")
```

Comment:
Because there's only two options, simplepush, and soon, webpush. External routers must use the RESTful HTTP API.